shared/src/commonMain/kotlin/com/karumi/gallery/domain/PhotosFlow.kt

```diff
@@ -0,0 +1,32 @@
+package com.karumi.gallery.domain
+
+import com.karumi.gallery.data.PhotosApiClient
+import com.karumi.gallery.model.Photos
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+import kotlin.time.Duration
+import kotlin.time.seconds
+
+class PhotosFlow(private val photosApiClient: PhotosApiClient) {
+
+  private companion object {
+    val updateInterval = 5.seconds
+  }
+
+  suspend fun get(): Flow<Photos> =
+    timer(repeatEvery = updateInterval)
+      .map { photosApiClient.getPhotos() }
```

Comment:
This is because you emit `Unit` but after that, you map it to the Photos you get from `.getPhotos()`, right?

Reply:
Yes, the timer does not emit anything of value, just happens to do it in a convenient interval :)